cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -3101,168 +3103,509 @@ func TestMutateGlobalPullSecret(t *testing.T) {
+							"qci-pull-through-cache-us-east-1-ci.apps.ci.l2s4.p1.openshiftapps.com": {
+								"auth": "supercool"
+							},
+							"c": {
+								"auth": "bar",
+								"email": "g"
+							}
+						}
+					}`),
+				},
+			},
+			expected: true,
+			mutatedSecret: &coreapi.Secret{
+				Data: map[string][]byte{
+					coreapi.DockerConfigJsonKey: []byte(`{
+						"auths": {
+							"a": {
+								"auth": "foo",
+								"email": "e"
+							},
+							"registry.ci.openshift.org": {
+								"auth": "cool"
+							},
+							"quay-proxy.ci.openshift.org": {
+								"auth": "supercool"
+							},
+							"quay.io/openshift/ci": {
+								"auth": "supercool"
+							},
+							"qci-pull-through-cache-us-east-1-ci.apps.ci.l2s4.p1.openshiftapps.com": {
+								"auth": "supercool"
+							}
+						}
+					}`),
+				},
+			},
+		},
+		{
+			name: "bad case: multiple expired auth's are replaced",
+			original: &coreapi.Secret{
+				Data: map[string][]byte{
+					coreapi.DockerConfigJsonKey: []byte(`{
+						"auths": {
+							"a": {
+								"auth": "foo",
+								"email": "e"
+							},
+							"registry.ci.openshift.org": {
+								"auth": "expired"
+							},
+							"quay-proxy.ci.openshift.org": {
+								"auth": "supercool"
+							},
+							"quay.io/openshift/ci": {
+								"auth": "expired"
+							},
+							"qci-pull-through-cache-us-east-1-ci.apps.ci.l2s4.p1.openshiftapps.com": {
+								"auth": "expired"
+							}
+						}
+					}`),
+				},
+			},
+			secret: &coreapi.Secret{
+				Data: map[string][]byte{
+					coreapi.DockerConfigJsonKey: []byte(`{
+						"auths": {
+							"a": {
+								"auth": "foo",
+								"email": "e"
+							},
+							"registry.ci.openshift.org": {
+								"auth": "cool"
+							},
+							"quay-proxy.ci.openshift.org": {
+								"auth": "supercool"
+							},
+							"quay.io/openshift/ci": {
+								"auth": "supercool"
+							},
+							"qci-pull-through-cache-us-east-1-ci.apps.ci.l2s4.p1.openshiftapps.com": {
+								"auth": "supercool"
+							},
+							"c": {
+								"auth": "bar",
+								"email": "g"
+							}
+						}
+					}`),
+				},
+			},
+			expected: true,
+			mutatedSecret: &coreapi.Secret{
+				Data: map[string][]byte{
+					coreapi.DockerConfigJsonKey: []byte(`{
+						"auths": {
+							"a": {
+								"auth": "foo",
+								"email": "e"
+							},
+							"registry.ci.openshift.org": {
+								"auth": "cool"
+							},
+							"quay-proxy.ci.openshift.org": {
+								"auth": "supercool"
+							},
+							"quay.io/openshift/ci": {
+								"auth": "supercool"
+							},
+							"qci-pull-through-cache-us-east-1-ci.apps.ci.l2s4.p1.openshiftapps.com": {
+								"auth": "supercool"
+							}
+						}
+					}`),
+				},
+			},
+		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			actual, actualErr := mutateGlobalPullSecret(tc.original, tc.secret)
-			if diff := cmp.Diff(actualErr, tc.expectedErr, testhelper.EquateErrorMessage); diff != "" {
```

Comment:
Why do we change this section?

Reply:
when the slices of registries are not sorted the diff provided is different. Using `cmpopts.SortedSlices()` was also an option but assert seemed more convenient

Reply:
Do you mind doing it with another PR?
I would like to keep the change minimal so that the review is easier and the chance of screwing thing up is smaller, in my opinion.